client/app/components/dashboards/widget.js

```diff
@@ -77,18 +77,20 @@ function DashboardWidgetCtrl($location, $uibModal, $window, $rootScope, Events,
 
   this.hasParameters = () => this.widget.query.getParametersDefs().length > 0;
 
-  this.editParameterMappings = () => {
-    $uibModal.open({
-      component: 'editParameterMappingsDialog',
-      resolve: {
-        dashboard: this.dashboard,
-        widget: this.widget,
-      },
-      size: 'lg',
-    }).result.then(() => {
-      this.localParameters = null;
-      $rootScope.$broadcast('dashboard.update-parameters');
-    });
+  this.mappingsDialogOpened = false;
+
+  this.openMappingsDialog = () => {
+    this.mappingsDialogOpened = true;
+  };
+
+  this.closeMappingsDialog = () => {
+    this.mappingsDialogOpened = false;
+    $rootScope.$applyAsync();
```

Comment:
No need to call `$applyAsync` on root scope - it will trigger too much checks. Inject `$scope` into controller and use `$scope.$applyAsync`

Reply:
Since reverting to uibmodal api, no need for these no more :)